src/core/raster/qgsrasterlayer.cpp

```diff
@@ -144,6 +144,14 @@ QgsRasterLayer::~QgsRasterLayer()
   // Note: provider and other interfaces are owned and deleted by pipe
 }
 
+QgsRasterLayer *QgsRasterLayer::clone( bool deep ) const
+{
+  QgsRasterLayer *layer = new QgsRasterLayer( source(), originalName(), mProviderKey );
+  QgsMapLayer::clone( layer, deep );
+  layer->setRenderer( renderer()->clone() );
+  return layer;
```

Comment:
The amount of stuff being cloned also feels very short - there is a whole raster pipeline that may need to be copied rather than just the renderer. Best to check all the private members and see if they get cloned.